src/versionbits.cpp

```diff
@@ -73,7 +73,7 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
     assert(cache.count(pindexPrev));
     ThresholdState state = cache[pindexPrev];
 
-    int nStartHeight = calculateStartHeight(pindexPrev, state, nPeriod, cache);
+    int nStartHeight = vToCompute.empty() ? 0 : calculateStartHeight(pindexPrev, state, nPeriod, cache);
```

Comment:
I think we could simply bail out early instead. That would be easier to read/understand imo.
```suggestion
    if (vToCompute.empty()) return state;

    int nStartHeight = calculateStartHeight(pindexPrev, state, nPeriod, cache);
```